spec/unit/provider/package/yum_spec.rb

```diff
@@ -153,14 +153,14 @@
         allow(@yum_cache).to receive(:yum_binary=).with("yum")
         @provider = Chef::Provider::Package::Yum.new(@new_resource, @run_context)
         @provider.load_current_resource
-        expect(@provider.new_resource.package_name).to eq("testing")
+        expect(@provider.new_resource.package_name).to eq("testing.noarch")
```

Comment:
i suspect that this indicates you've actually broken a use case, but i proving it, without taking a week to create a bunch of functional end-to-end tests is going to be difficult.